src/osd/PrimaryLogPG.cc

```diff
@@ -6400,6 +6400,7 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	        oi.size - op.extent.truncate_size);
 	      ctx->modified_ranges.union_of(trim);
 	      ctx->clean_regions.mark_data_region_dirty(op.extent.truncate_size, oi.size - op.extent.truncate_size);
+	      oi.clear_data_digest();
```

Comment:
when truncate smaller, need trim some data, we need clear_data_digest.
because data_digest is calced by old data.
if we truncate bigger, i think need clear_data_digest too.
because data size is updated, data_digest is old too.